packages/golden-layout/src/items/Component.ts

```diff
@@ -27,6 +27,16 @@ export default class Component extends AbstractContentItem {
     config: ComponentConfig,
     parent: AbstractContentItem
   ) {
+    // Extend before super so the AbstractContentItem defualts aren't applied first
+    Object.entries(
+      layoutManager.config.settings.defaultComponentConfig ?? {}
+    ).forEach(([k, v]) => {
+      // @ts-ignore
+      if (config[k] === undefined) {
+        // @ts-ignore
+        config[k] = v;
+      }
+    });
```

Comment:
Annoying that `Object.assign` doesn't have something that does this, skipping `undefined`. Oh well.